arxiv/release/README.md

```diff
@@ -0,0 +1,65 @@
+# Dealing with versions from git tags
+
+This contains functions for dealing with versions from git tags. They
+are intended for use in continuous integration tools.
+
+The general idea is to get the version from the git tag when making a
+package or docker image. Out side of these cases the version is from
+`git describe` to provide an accurate description of the version and
+changes.
+
+## Goals
+
+1. Reduce human error. We have had problems with making a new version
+   of a repo but missing the task of setting a new version in files
+   like arxiv/base/config.py or in setup.py.
+2. Have one source of the version. The git tag is saved to a
+   version.py file and that is used in setup.py and
+   arxiv/SOMEPKG/version.py. 
+3. Versions on web pages should be correct. On non-production web
+   pages use 0.15.8 if loaded from a pypi package or docker image, or
+   something like 0.15.8-48-dirty if from a git repo.
```

Comment:
I couldn't get a good handle on what you meant by this goal. Could it be rephrased or expanded?

Reply:
Right now in the submission-ui there is a version on the Manage Submissions page of "0.1.4 2019-05-04". This is not really correct.

Once this system is setup in submisison-ui that could be "heads/ARXIVNG-2970-dirty"  or "0.1.5rc1-54-g36459ae-dirty" which means 54 commits ahead of 0.1.5rc1 and the working tree is uncommitted changes.

There is a distinction here between production and non-prod. This is because production should always be from docker images or pypi packages but Non-prod could be from those or from git repos.

I don't currently have a good rephrase of this goal.